src/metabase/pulse/render/body.clj

```diff
@@ -272,7 +276,7 @@
   [col-settings col]
   (-> (m/map-keys (fn [k] (-> k name (str/replace #"-" "_") keyword)) col-settings)
       (backfill-currency)
-      (u/update-if-exists :date_style update-date-style (:unit col))))
+      (u/update-if-exists :date_style update-date-style (:unit col) col-settings)))
```

Comment:
`col-settings` here has a shape like this: 

```clojure
#:metabase.shared.models.visualization-settings{:date-style "YYYY/M/D"
                                                :date-separator "-"
                                                :date-abbreviate true
                                                :time-style "HH:mm"
                                                :time-enabled "minutes"}
```

Which is a map of viz-settings that correspond to data options a user has specified in the visualization UI. It gets passed in to `update-date-style` which I modified to change the date style format in a way that respects those settings, where previously it did not.

Reply:
This is something that could be nice as a comment in the code itself. The next person to touch this is never going to see this info and is going to have to rediscover it the same way you did